packages/wallet/frontend/src/pages/index.tsx

```diff
@@ -16,10 +16,42 @@ import { userService } from '@/lib/api/user'
 import type { NextPageWithLayout } from '@/lib/types/app'
 import { useOnboardingContext } from '@/lib/context/onboarding'
 import { useEffect } from 'react'
+import { io, Socket } from 'socket.io-client'
 
 type HomeProps = InferGetServerSidePropsType<typeof getServerSideProps>
 
 const HomePage: NextPageWithLayout<HomeProps> = ({ accounts, user }) => {
+  let socket: Socket
+
+  useEffect(() => {
+    // Access cookies
+
+    // Connect to the Socket.IO server
+    socket = io(process.env.NEXT_PUBLIC_BACKEND_URL ?? '', {
+      query: {
+        email: user.email
```

Comment:
instead of passing the email, the authorization token should be set in request and the user info will be extracted from there on authorization